django_usaid_clinicfinder/clinicfinder/models.py

```diff
@@ -64,4 +67,33 @@ class LookupPointOfInterest(HStoreModel):
     search = hstore.DictionaryField()
     response = hstore.DictionaryField()
     location = djangomodels.ForeignKey(
-        LookupLocation, related_name='lookup_location')
+        LookupLocation, related_name='lookup_location',
+        blank=True, null=True)
+
+
+class LBSRequest(HStoreModel):
+
+    """
+    Inbound request for LBS lookup. Triggers LBS API call.
+    """
+    created_at = djangomodels.DateTimeField(auto_now_add=True)
+    updated_at = djangomodels.DateTimeField(auto_now=True)
+    search = hstore.DictionaryField()
+    response = hstore.DictionaryField(blank=True, null=True)
+    pointofinterest = djangomodels.ForeignKey(
+        LookupPointOfInterest, related_name='pointofinterest')
+
+
+# Make sure new LBS Requests tasks are run via Celery
+@receiver(post_save, sender=LBSRequest)
+def fire_lbs_task_if_new(sender, instance, created, **kwargs):
+    if created:
+        lbs_lookup.delay(instance.id)
+
+
+@receiver(post_save, sender=LookupPointOfInterest)
+def fire_location_finder_task_if_complete(sender, instance, created, **kwargs):
+    # Lookup locaction in place and results not in place already
+    if instance.location != None and "results" not in instance.response:
+        # find match and prepare response
+        location_finder.delay(instance.id)
```

Comment:
I'm not sure I'm happy about hooking up SMS sending to database actions. It seems less surprising to me to have to explicitly send SMSes and to chain these tasks together using Celery's task chaining? So something like:
```
task = lookup_lsb.s() | location_finder.s() | location_sender.s()
task.delay()
```
(hopefully I got all the celery task mask roughly right)